### ISSUE TITLE
Query with window expression and join fails with "Field is not streamable"

### ISSUE THREAD

**Issue by MarkusH** [crate/crate#10189]
**CrateDB version**: 4.1.7

**Problem description**:

The following query fails:

```sql
SELECT
  date_trunc('day', ts) AS "time",
  SUM(ts - prev_ts)
FROM
  (
    SELECT
      t.ts,
      lag(t.ts) OVER (PARTITION BY t.cluster_id ORDER BY t.ts) AS prev_ts
    FROM
      tracking AS t
    INNER JOIN
      clusters AS c
      ON t.cluster_id = c.id
    INNER JOIN
      projects AS p
      ON c.project_id = p.id
    WHERE
      t.ts >= '2020-06-07T15:33:42.965Z'
      AND t.ts <= date_trunc('day', '2020-07-07T15:33:42.965Z')
      AND p.organization_id NOT IN ('a', 'b')
  ) AS t1
GROUP BY "time"
ORDER BY "time"
;
```

```java
SQLActionException: 400 Bad Request 4004 UnsupportedFeatureException: Field is not streamable
        at io.crate.expression.symbol.Field.writeTo(Field.java:102)
        at io.crate.expression.symbol.Symbols.toStream(Symbols.java:124)
        at io.crate.expression.symbol.Symbols.toStream(Symbols.java:108)
        at io.crate.expression.symbol.Function.writeTo(Function.java:150)
        at io.crate.expression.symbol.WindowFunction.writeTo(WindowFunction.java:79)
        at io.crate.expression.symbol.Symbols.toStream(Symbols.java:124)
        at io.crate.execution.dsl.projection.WindowAggProjection.writeTo(WindowAggProjection.java:150)
        at io.crate.execution.dsl.projection.Projection.toStream(Projection.java:67)
        at io.crate.execution.dsl.phases.AbstractProjectionsPhase.writeTo(AbstractProjectionsPhase.java:146)
        at io.crate.execution.dsl.phases.MergePhase.writeTo(MergePhase.java:181)
        at io.crate.execution.dsl.phases.ExecutionPhases.toStream(ExecutionPhases.java:56)
        at io.crate.execution.dsl.phases.NodeOperation.writeTo(NodeOperation.java:74)
        at io.crate.execution.jobs.transport.JobRequest.writeTo(JobRequest.java:103)
        at org.elasticsearch.transport.TcpTransport.buildMessage(TcpTransport.java:1004)
        at org.elasticsearch.transport.TcpTransport.sendRequestToChannel(TcpTransport.java:855)
        at org.elasticsearch.transport.TcpTransport$NodeChannels.sendRequest(TcpTransport.java:371)
        at org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:602)
        at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:517)
        at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:505)
        at io.crate.execution.support.Transports.sendRequest(Transports.java:94)
        at io.crate.execution.support.Transports.sendRequest(Transports.java:103)
        at io.crate.execution.jobs.transport.TransportJobAction.execute(TransportJobAction.java:79)
        at io.crate.execution.engine.JobLauncher.sendJobRequests(JobLauncher.java:319)
        at io.crate.execution.engine.JobLauncher.setupTasks(JobLauncher.java:244)
        at io.crate.execution.engine.JobLauncher.execute(JobLauncher.java:154)
        at io.crate.planner.operators.LogicalPlanner.executeNodeOpTree(LogicalPlanner.java:438)
        at io.crate.planner.operators.LogicalPlanner.doExecute(LogicalPlanner.java:410)
        at io.crate.planner.operators.LogicalPlanner.execute(LogicalPlanner.java:389)
        at io.crate.planner.operators.LogicalPlan.executeOrFail(LogicalPlan.java:174)
        at io.crate.planner.Plan.execute(Plan.java:74)
        at io.crate.action.sql.Session.singleExec(Session.java:597)
        at io.crate.action.sql.Session.exec(Session.java:465)
        at io.crate.action.sql.Session.sync(Session.java:444)
        at io.crate.rest.action.SqlHttpHandler.executeSimpleRequest(SqlHttpHandler.java:272)
        at io.crate.rest.action.SqlHttpHandler.handleSQLRequest(SqlHttpHandler.java:207)
        at io.crate.rest.action.SqlHttpHandler.channelRead0(SqlHttpHandler.java:120)
        at io.crate.rest.action.SqlHttpHandler.channelRead0(SqlHttpHandler.java:84)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
        at io.crate.protocols.http.HttpBlobHandler.channelRead0(HttpBlobHandler.java:166)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
        at io.crate.protocols.http.HttpAuthUpstreamHandler.handleHttpChunk(HttpAuthUpstreamHandler.java:134)
        at io.crate.protocols.http.HttpAuthUpstreamHandler.channelRead0(HttpAuthUpstreamHandler.java:83)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
        at org.elasticsearch.http.netty4.cors.Netty4CorsHandler.channelRead(Netty4CorsHandler.java:85)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:321)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:295)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:355)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:377)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:363)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by mfussenegger** [comment]
@MarkusH Could you also share the schema of the tables involved in the query (Or a subset of it to have the used columns) ?

**Comment by MarkusH** [comment]
I surely can, @mfussenegger!

```sql
CREATE TABLE "tracking" (
   "ts" TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP,
   "cluster_id" TEXT NOT NULL
)
CLUSTERED INTO 1 SHARDS
WITH (
   "column_policy" = 'strict',
   "number_of_replicas" = '5',
   "write.wait_for_active_shards" = '1'
);


CREATE TABLE "clusters" (
   "id" TEXT,
   "project_id" TEXT,
   PRIMARY KEY ("id")
)
CLUSTERED INTO 1 SHARDS
WITH (
   "column_policy" = 'strict',
   "number_of_replicas" = '5',
   "write.wait_for_active_shards" = '1'
);


CREATE TABLE "projects" (
   "id" TEXT,
   "organization_id" TEXT,
   PRIMARY KEY ("id")
)
CLUSTERED INTO 1 SHARDS
WITH (
   "column_policy" = 'strict',
   "number_of_replicas" = '5',
   "write.wait_for_active_shards" = '1'
);
```

**Comment by mfussenegger** [comment]
Fixed this with https://github.com/crate/crate/pull/10191 - which will be available with 4.2.1
